packages/create-cloudflare/src/common.ts

```diff
@@ -135,32 +136,63 @@ export const setupProjectDirectory = (args: C3Args) => {
 	return { name: pathBasename, path };
 };
 
-export const offerToDeploy = async (ctx: PagesGeneratorContext) => {
+export const offerToDeploy = async (ctx: C3Context) => {
 	startSection(`Deploy with Cloudflare`, `Step 3 of 3`);
 
+	// Coerce no-deploy if it isn't possible (i.e. if its a worker with any bindings)
+	if (!(await isDeployable(ctx))) {
+		ctx.args.deploy = false;
+	}
+
 	const label = `deploy via \`${quoteShellArgs([
 		npm,
 		"run",
-		...(ctx.framework?.config.deployCommand ?? ["deploy"]),
+		...(ctx.template.deployCommand ?? ["deploy"]),
 	])}\``;
 
-	ctx.args.deploy = await processArgument(ctx.args, "deploy", {
+	const shouldDeploy = await processArgument(ctx.args, "deploy", {
 		type: "confirm",
 		question: "Do you want to deploy your application?",
 		label,
 		defaultValue: C3_DEFAULTS.deploy,
 	});
 
-	if (!ctx.args.deploy) return;
+	if (!shouldDeploy) {
+		return false;
+	}
+
+	// initialize a deployment object in context
+	ctx.deployment = {};
 
 	const loginSuccess = await wranglerLogin();
 	if (!loginSuccess) return;
 
 	await chooseAccount(ctx);
+
+	return true;
 };
 
-export const runDeploy = async (ctx: PagesGeneratorContext) => {
-	if (ctx.args.deploy === false) return;
+/**
+ * Determines if the current project is deployable.
+ *
+ * Since C3 doesn't currently support a way to automatically provision the resources needed
+ * by bindings, templates that have placeholder bindings will need some adjustment by the project author
+ * before they can be deployed.
+ */
+const isDeployable = async (ctx: C3Context) => {
+	if (ctx.template.platform === "pages") {
+		return true;
+	}
+
+	const wranglerToml = await readWranglerToml(ctx);
+	if (wranglerToml.match(/(?<!#\s*)bindings?\s*=.*/m)) {
```

Comment:
Is it worth logging a message here so that the user understands why they did not get asked to deploy their new project?